### DISCUSSION TITLE
Adding non-metadata items to <head> in app directory

### DISCUSSION THREAD

**Discussion by controversial** [vercel/next.js#46816] [Help]
### Summary

With the deprecation of `head.js`, how can I add (non-metadata) elements to the `<head>` of the document for a specific set of pages?

I sometimes want to inject a small stylesheet into the head on some specific page, for example to add styles to `body` based on some data fetching. In the `pages` directory, I could do something like:

```tsx
...

<Head>
  <style>{`
    :root {
      --some-variable: ${valueFromServerSideProps};
    }
    body {
      /* root-level styles that only apply to this page, e.g. page background color */
    }
  `}</style>
</Head>

...
```
Now, with the deprecation of `head.js`, it’s not clear to me how to add non-metadata tags to the <Head> of a document outside of the root layout.

The [section on “Unsupported metadata”](https://beta.nextjs.org/docs/api-reference/metadata#unsupported-metadata) recommends “render the tag in the layout or page itself” for a few of the listed uses:
<img width="500" alt="Screenshot 2023-03-06 at 2 59 12 AM" src="https://user-images.githubusercontent.com/10377391/223051381-03b7a63f-78e9-4880-92f9-708eb3714f8c.png">

but without the use of `head.js`, it’s not clear to me *how* I can render markup in `<head>` from a “layout or page” that isn’t the root layout. What am I missing?

**Comment by jack-pallot** [comment]
Similar to this, some CMS's will provide a block of HTML to render the entire head section. Without page.js, this no longer seems possible to do. Perhaps there needs to be a "raw" or "extends" property on the metadata object to handle situations like this?

```
const metadata = {
  raw: `${fullHtml}`
}
```

**Comment by maxswjeon** [comment]
I second this, as I am trying to develop a new CMS with Next.js.
